server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/pipelines_controller.rb

```diff
@@ -119,6 +126,18 @@ def check_if_pipeline_by_same_name_already_exists
         end
       end
 
+      def check_for_pipeline_dependency
+        can_be_deleted_message = LocalizedMessage.string("CAN_DELETE_PIPELINE")
+        pipelines_can_be_deleted = pipeline_config_service.canDeletePipelines()
+        actual_message = pipelines_can_be_deleted[CaseInsensitiveString.new(params[:name])].message
+
+        if(!can_be_deleted_message.equals(actual_message))
```

Comment:
Ouch, this seems like a dangerous thing to do! relying on a string message to imply deletion is not a great idea.

Consider using `CanDeleteResult#canDelete` instead?